### PR TITLE
Ft wishlist 187300161

### PR DESCRIPTION
**Pull request by Mukunzijames**
What does this PR do?

this PR for wishlist functionality: Added feature to view, add, and delete items. Enhanced user experience with seamless management of wishlisted items, ensuring smoother navigation and interaction.

Description of Task to be completed?
The task involves the following key functionalities:

- Validate that the buyer is logged in.
- Add the item to the buyer's wishlist.
- Retrieve the product details from the database.
- Store the updated wishlist in the buyer's account.
- Return a confirmation message to the frontend."

What are the relevant pivotal tracker stories?
Story ID:#187300161

### CODE REVIEW

File: src/controllers/wishlistController.ts [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+export const getWishlist = async (req: Request, res: Response) => {
+  const { id, role } = req.user as UserInterface;
+
+  try {
+    let wishlist;
+    if (role === 'seller') {
+      const products = await Database.Product.findAll({
+        where: { sellerId: id },
+      });
+      const productIds = products.map(product => product.id);
```
Comment (fab-ryan) [inline src/controllers/wishlistController.ts]: you can get only ids using select commands

File: src/controllers/wishlistController.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+      wishlist = await Database.Wishlist.findAll({
+        where: { productId: productIds },
+        include: [{ model: Database.Product, as: 'product' }],
+      });
+    } else {
+      wishlist = await Database.Wishlist.findAll({
+        where: { userId: id },
+        include: [{ model: Database.Product, as: 'product' }],
```
Comment (fab-ryan) [inline src/controllers/wishlistController.ts]: make as products not product

File: src/database/models/wishlist.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+  /**
+   * Overrides the default toJSON method to exclude the password field.
+   * @returns {Object} An object representing the user, excluding the password.
+   */
+  toJSON() {
+    return {
+      id: this.id,
+      userId: this.userId,
+      productId: this.productId,
```
Comment (fab-ryan) [inline src/database/models/wishlist.ts]: the product will even be returned?

File: src/middleware/authentication.ts [hunk 4]
```diff
@@ -59,7 +59,14 @@ export const isSeller = (req: Request, res: Response, next: NextFunction) => {
 };
 export const isAdmin = (req: Request, res: Response, next: NextFunction) => {
   if (!req.user || !('role' in req.user) || req.user.role !== 'admin') {
-    return sendResponse(res, 401, null, 'Not authorized! User should be admin');
+    return sendResponse(res, 403, null, 'Not authorized! User should be admin');
```
Comment (fab-ryan) [inline src/middleware/authentication.ts]: only the Not authorized is enuogh donn't specify the role

**Review by fab-ryan — CHANGES_REQUESTED** [review]
work on those comments

**Review by musabehonore — APPROVED** [review]

**Review by Nadinefiona — APPROVED** [review]

**Review by fab-ryan — APPROVED** [review]
good work

**Pull request merged**